dnf/conf/config.py

```diff
@@ -229,6 +229,13 @@ def __init__(self, section='main', parser=None):
         self._config.cachedir().set(PRIO_DEFAULT, cachedir)
         self._config.logdir().set(PRIO_DEFAULT, logdir)
 
+        # track list of temporary files created
+        self.tempfiles = []
+
+    def __del__(self):
+        for file_name in self.tempfiles:
+            os.unlink(file_name)
+
```

Comment:
Why make this a separate commit? The cleanup goes with the main feature I think...

Reply:
Just because I added it later. But you are right and I will squash the commits.